infer/src/base/ProcessPool.ml

```diff
@@ -370,52 +418,66 @@ let fork_child ~child_prologue ~slot (updates_r, updates_w) ~f ~epilogue =
   | `In_the_child ->
       Unix.close updates_r ;
       Unix.close to_child_w ;
-      (* Pin to a core. [setcore] does the modulo <number of cores> for us. *)
-      Utils.set_best_cpu_for slot ;
-      ProcessPoolState.in_child := true ;
       ProcessPoolState.reset_pid () ;
-      child_prologue () ;
       let updates_oc = Unix.out_channel_of_descr updates_w in
-      let send_to_parent (message : 'b worker_message) = marshal_to_pipe updates_oc message in
-      let send_final (final_message : 'a final_worker_message) =
-        marshal_to_pipe updates_oc final_message
-      in
-      (* Function to send updates up the pipe to the parent instead of directly to the task
-         bar. This is because only the parent knows about all the children, hence it's in charge of
-         actually updating the task bar. *)
-      let update_status t status =
-        match Config.progress_bar with
-        | `Quiet | `Plain ->
-            ()
-        | `MultiLine ->
-            let status =
-              (* Truncate status if too big: it's pointless to spam the status bar with long status, and
-                 also difficult to achieve technically over pipes (it's easier if all the messages fit
-                 into a buffer of reasonable size). *)
-              if String.length status > 100 then String.subo ~len:100 status ^ "..." else status
-            in
-            send_to_parent (UpdateStatus (slot, t, status))
-      in
-      ProcessPoolState.update_status := update_status ;
       let orders_ic = Unix.in_channel_of_descr to_child_r in
-      let receive_from_parent () =
-        PerfEvent.log (fun logger ->
-            PerfEvent.log_begin_event logger ~categories:["sys"] ~name:"receive from pipe" () ) ;
-        let x = Marshal.from_channel orders_ic in
-        PerfEvent.(log (fun logger -> log_end_event logger ())) ;
-        x
-      in
-      child_loop ~slot send_to_parent send_final receive_from_parent ~f ~epilogue ~prev_result:None ;
-      Out_channel.close updates_oc ;
-      In_channel.close orders_ic ;
-      Epilogues.run () ;
-      Stdlib.exit 0
+      child slot ~f ~child_prologue ~epilogue ~updates_oc ~orders_ic
   | `In_the_parent pid ->
       Unix.close to_child_r ;
       Unix.close updates_w ;
       {pid; down_pipe= Unix.out_channel_of_descr to_child_w}
 
 
+(** Data marshalled to describe what a non-forked child must do. *)
```

Comment:
s/non-forked/spawned/ for clarity, and may add "by [spawn_child] below"?

Reply:
Done.

Reply:
Can be resolved.